atlasdb-commons/src/test/java/com/palantir/util/CachedTransformingSupplierTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Test;
+
+public class CachedTransformingSupplierTest {
+
+    @Test
+    public void test() {
+        AtomicInteger inputValue = new AtomicInteger();
+        AtomicInteger outputValue = new AtomicInteger();
+
+        CachedTransformingSupplier<Integer, Integer> supplier =
+                new CachedTransformingSupplier<>(inputValue::get, input -> input + outputValue.incrementAndGet());
+
+        assertThat(supplier.get()).isEqualTo(1);
+        assertThat(supplier.get()).isEqualTo(1);
```

Comment:
Is it worth verifying the function has only been invoked once?

```suggestion
        assertThat(supplier.get()).isEqualTo(1);
        assertThat(outputValue).hasValue(1);
```